bitsandbytes/functional.py

```diff
@@ -15,10 +15,29 @@
 from functools import reduce  # Required in Python 3
 from typing import Tuple, Any, Dict
 from torch import Tensor
+
+from warnings import warn
+from .cextension import COMPILED_WITH_CUDA
 from bitsandbytes.utils import pack_dict_to_tensor, unpack_tensor_to_dict
 
-from .cextension import COMPILED_WITH_CUDA, lib
 
+# CUDA specific lib
+if COMPILED_WITH_CUDA:
+    from .cextension import lib
+
+from bitsandbytes.device_setup.cpu.main import is_ipex_cpu_available
+from bitsandbytes.device_setup.xpu.main import is_ipex_xpu_available
+if not is_ipex_cpu_available():
+    warn(
+        "Intel Extension for PyTorch CPU/XPU supports are not available."
+        "Please refer to https://intel.github.io/intel-extension-for-pytorch/ for installation."
+        )
+else:
+    if not is_ipex_xpu_available():
+        warn(
+            "Intel Extension for PyTorch CPU support is available, while XPU is not."
+            )
+    import intel_extension_for_pytorch as ipex
```

Comment:
Remove all the IPEX checking in this PR since we focus on device abstraction for 8bits/4bits and no real usage in this PR. And will add them in 8bits/4bits CPU/XPU functionality PRs